test/benchmark_http_client_test.cc

```diff
@@ -121,14 +132,25 @@ class BenchmarkClientHttpTest : public Test {
     decoders_.clear();
     dispatcher_->run(Envoy::Event::Dispatcher::RunType::Block);
     EXPECT_EQ(0, inflight_response_count);
+    // If we have no expectations, then we don't test.
+    if (header_expectations.size() != 0) {
```

Comment:
it's not clear from the method signature that empty header_expectation would simply ignore the check instead of validating that the returned header were empty

Reply:
Add clarifying comments on the method, not sure if there's a better way to do that.